include/librealsense2/dds/dds-participant.cpp

```diff
@@ -100,12 +113,17 @@ void dds_participant::init( dds_domain_id domain_id, std::string const & partici
     DomainParticipantQos pqos;
     pqos.name( participant_name );
 
-    // Indicates for how much time should a remote DomainParticipant consider the local
-    // DomainParticipant to be alive.
+    // Indicates for how much time should a remote DomainParticipant consider the local DomainParticipant to be alive.
     pqos.wire_protocol().builtin.discovery_config.leaseDuration = { 10, 0 };  // [sec,nsec]
 
-    _participant = DDS_API_CALL(DomainParticipantFactory::get_instance()->create_participant( domain_id, pqos, _domain_listener.get() ));
-    
+    // Listener will call DataReaderListener::on_data_available for a specific reader,
+    // not SubscriberListener::on_data_on_readers for any reader
+    StatusMask par_mask = StatusMask::all() >> StatusMask::data_on_readers();
```

Comment:
Can you elaborate the comment? I didn't understand the mask logic

Reply:
Wow what horrible API...
`mask >> bits` is the same as saying `mask &= ~bits`.
I'd prefer see the latter. Maybe that's what we should write?

Reply:
I found this note in the DDS docs:
>**Note**: The on_data_on_readers() callback intercepts messages before on_data_available(). This implies that if DomainParticipantListener is enabled, users should take into account that by default the listener uses StatusMask::all(). As the callback entity hierarchy is kept, the on_data_on_readers() is going to be called in this case. If an application wants to use on_data_available() instead, the corresponding bit of StatusMask should be disabled.

Maybe we should embed this note in the comment?

Reply:
I'll add a link to the DDS docs note

Reply:
Done